Issue#30- initial implementation

ashtkumar2023 opened the pull request:
Hi @robe070 , Kindly review .Thanks
**PR Description:** 

-  Issue# 30 intial commit.
-  It contains the "Search <1>" results tested for the below except Integer 1,2.. 8 routines.
   #Com_Owner.mthSearch_By_Signed_Decimals( #LF_Test_Data #this_rc )
   #Com_Owner.mthSearch_By_Signed( #LF_Test_Data #this_rc )
   #Com_Owner.mthSearch_By_Decimal_Decimals( #LF_Test_Data #this_rc )
   #Com_Owner.mthSearch_By_Decimal( #LF_Test_Data #this_rc ).
-  Implementation for Integer1 through Integer8 are also commented for now.
-  Search<2> through Search<8> is also commented for now. 

 **NOTE** :I will be testing the remaining implementation , sharing another PR asap containing all.
 
 **Test data and results below.**
   2023-12-25 10:59:14 + (4) - Calling Function: V57033F
    2023-12-25 10:59:14 - V57033F. Start - Generic Numeric Search on LV
    2023-12-25 10:59:14 -  Delete Data. 1 set of key records deleted
    2023-12-25 10:59:14 -  Insert Data. 640 / 640 records inserted
    2023-12-25 10:59:14 -  <1> Search Test: '100.00'.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: S(7,2). *FORWARDS mthSearch_By_Signed_Decimals
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L157033FND. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L157033FPD. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: S(7,2). *BACKWARDS mthSearch_By_Signed_Decimals
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L157033FND. Rows = 321 Expected = 321.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L157033FPD. Rows = 321 Expected = 321.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: S(7,2). *FORWARDS DESCENDING mthSearch_By_Signed_Decimals
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L57033FNDD. Rows = 321 Expected = 321.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L57033FPDD. Rows = 321 Expected = 321.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: S(7,2). *BACKWARDS DESCENDING mthSearch_By_Signed_Decimals
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L57033FNDD. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L57033FPDD. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  *** SELECT EXACT Key: S(7,2). *FORWARDS mthFunc_Search_Exact_Signed_Decimals
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L157033FND. Rows = 1 Expected = 1.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L157033FPD. Rows = 1 Expected = 1.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: S(7,0). *FORWARDS mthFunc_Search_Generic_Signed
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FND. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: S(7,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FN. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FPD. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: INTEGER 1. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI1. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: INTEGER 2. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI2. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: INTEGER 4. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI4. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: INTEGER 8. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI8. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: S(7,0). *BACKWARDS mthFunc_Search_Generic_Signed
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FND *BACKWARDS. Rows = 321.  Expected = 321
    2023-12-25 10:59:14 -  Key: S(7,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FN *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FPD *BACKWARDS. Rows = 321.  Expected = 321
    2023-12-25 10:59:14 -  Key: INTEGER 1. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI1 *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-25 10:59:14 -  Key: INTEGER 2. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI2. Rows = 336 Expected = 336.
    2023-12-25 10:59:14 -  Key: INTEGER 4. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI4. Rows = 336 Expected = 336.
    2023-12-25 10:59:14 -  Key: INTEGER 8. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI8. Rows = 336 Expected = 336.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: S(7,0). *FORWARDS DESCENDING mthFunc_Search_Generic_Signed
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FNDD. Rows = 642 Expected = 642.
    2023-12-25 10:59:14 -  Key: S(7,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FNN. Rows = 672 Expected = 672.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FPDD. Rows = 642 Expected = 642.
    2023-12-25 10:59:14 -  Key: INTEGER 1. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI1D. Rows = 672 Expected = 672.
    2023-12-25 10:59:14 -  Key: INTEGER 2. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI2D. Rows = 672 Expected = 672.
    2023-12-25 10:59:14 -  Key: INTEGER 4. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI4D. Rows = 672 Expected = 672.
    2023-12-25 10:59:14 -  Key: INTEGER 8. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI8D. Rows = 672 Expected = 672.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: S(7,0). *BACKWARDS DESCENDING mthFunc_Search_Generic_Signed
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FNDD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  Key: S(7,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FND *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FPDD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  Key: INTEGER 1. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI1D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  Key: INTEGER 2. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI2D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  Key: INTEGER 4. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI4D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  Key: INTEGER 8. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI8D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  *** SELECT EXACT Key: S(7,0). *FORWARDS mthFunc_Search_Exact_Signed
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FND. Rows = 1 Expected = 1.
    2023-12-25 10:59:14 -  Key: S(7,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FN. Rows = 16 Expected = 16.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FPD. Rows = 1 Expected = 1.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FPA. Rows = 16 Expected = 16.
    2023-12-25 10:59:14 -  Key: INTEGER 1. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI1. Rows = 16 Expected = 16.
    2023-12-25 10:59:14 -  Key: INTEGER 2. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI2. Rows = 16 Expected = 16.
    2023-12-25 10:59:14 -  Key: INTEGER 4. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI4. Rows = 16 Expected = 16.
    2023-12-25 10:59:14 -  Key: INTEGER 8. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI8. Rows = 16 Expected = 16.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: Packed(15,4). *FORWARDS mthSearch_By_Packed_Decimals
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L157033FND. Rows = 0 Expected = 0.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L157033FPD. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: Packed(15,4). *BACKWARDS mthSearch_By_Packed_Decimals
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L157033FND. Rows = 0 Expected = 0.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L157033FPD. Rows = 321 Expected = 321.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: Packed(15,4). *FORWARDS DESCENDING mthSearch_By_Packed_Decimals
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L57033FNDD. Rows = 0 Expected = 0.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L57033FPDD. Rows = 321 Expected = 321.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: Packed(15,4). *BACKWARDS DESCENDING mthSearch_By_Packed_Decimals
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L57033FNDD. Rows = 0 Expected = 0.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L57033FPDD. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  *** SELECT EXACT Key: Packed(15,4). *FORWARDS mthFunc_Search_Exact_Packed_Decimals
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L157033FND. Rows = 1 Expected = 1.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-25 10:59:14 -  On LV L157033FPD. Rows = 1 Expected = 1.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: Packed(15,0). *FORWARDS mthFunc_Search_Generic_Decimal
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FND. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: S(7,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FN. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FPD. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FPA. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: INTEGER 1. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI1. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: INTEGER 2. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI2. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: INTEGER 4. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI4. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  Key: INTEGER 8. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI8. Rows = 320 Expected = 320.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: Packed(15,0). *BACKWARDS mthFunc_Search_Generic_Decimal
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FND *BACKWARDS. Rows = 321.  Expected = 321
    2023-12-25 10:59:14 -  Key: S(7,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FN *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FPD *BACKWARDS. Rows = 321.  Expected = 321
    2023-12-25 10:59:14 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FPA *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-25 10:59:14 -  Key: INTEGER 1. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI1 *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-25 10:59:14 -  Key: INTEGER 2. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI2 *BACKWARDS. Rows = 336 Expected = 336.
    2023-12-25 10:59:14 -  Key: INTEGER 4. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI4 *BACKWARDS. Rows = 336 Expected = 336.
    2023-12-25 10:59:14 -  Key: INTEGER 8. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI8 *BACKWARDS. Rows = 336 Expected = 336.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: Packed(15,0). *FORWARDS DESCENDING mthFunc_Search_Generic_Decimal
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FNDD *FORWARDS. Rows = 642 Expected = 642.
    2023-12-25 10:59:14 -  Key: S(7,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FNN *FORWARDS. Rows = 672 Expected = 672.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FPDD *FORWARDS. Rows = 642 Expected = 642.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FPD. Rows = 672 Expected = 672.
    2023-12-25 10:59:14 -  Key: INTEGER 1. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI1D. Rows = 672 Expected = 672.
    2023-12-25 10:59:14 -  Key: INTEGER 2. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI2D. Rows = 672 Expected = 672.
    2023-12-25 10:59:14 -  Key: INTEGER 4. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI4D. Rows = 672 Expected = 672.
    2023-12-25 10:59:14 -  Key: INTEGER 8. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI8D. Rows = 672 Expected = 672.
    2023-12-25 10:59:14 -  *** SELECT GENERIC Key: Packed(15,0). *BACKWARDS DESCENDING mthFunc_Search_Generic_Decimal
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FNDD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  Key: S(7,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FND *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FPDD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FPD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  Key: INTEGER 1. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI1D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  Key: INTEGER 2. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI2D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  Key: INTEGER 4. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI4D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  Key: INTEGER 8. Key Value:100
    2023-12-25 10:59:14 -  On LV L57033FI8D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-25 10:59:14 -  *** SELECT EXACT Key: Packed(15,0). *FORWARDS mthFunc_Search_Exact_Decimal
    2023-12-25 10:59:14 -  Key: S(7,2). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FND. Rows = 1 Expected = 1.
    2023-12-25 10:59:14 -  Key: S(7,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FN. Rows = 16 Expected = 16.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FPD. Rows = 1 Expected = 1.
    2023-12-25 10:59:14 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FPA. Rows = 16 Expected = 16.
    2023-12-25 10:59:14 -  Key: INTEGER 1. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI1. Rows = 16 Expected = 16.
    2023-12-25 10:59:14 -  Key: INTEGER 2. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI2. Rows = 16 Expected = 16.
    2023-12-25 10:59:14 -  Key: INTEGER 4. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI4. Rows = 16 Expected = 16.
    2023-12-25 10:59:14 -  Key: INTEGER 8. Key Value:100
    2023-12-25 10:59:14 -  On LV L157033FI8. Rows = 16 Expected = 16.
    2023-12-25 10:59:14 - V57033F. End
    2023-12-25 10:59:14 Completed with <OK>.

ashtkumar2023 committed:
Issue#30- initial implementation

ashtkumar2023 committed:
Moved the new code below in Search<1> as per feedback

ashtkumar2023 committed:
Included Integer1 and Integer2 TestResults in Search<1>

ashtkumar2023 commented:
**Test data and results for commit Id : a084913**
 2023-12-26 15:16:30 + (4) - Calling Function: V57033F
    2023-12-26 15:16:30 - V57033F. Start - Generic Numeric Search on LV
    2023-12-26 15:16:30 -  Delete Data. 1 set of key records deleted
    2023-12-26 15:16:30 -  Insert Data. 640 / 640 records inserted
    2023-12-26 15:16:30 -  <1> Search Test: '100.00'.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: S(7,2). *FORWARDS mthSearch_By_Signed_Decimals
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L157033FND. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L157033FPD. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: S(7,2). *BACKWARDS mthSearch_By_Signed_Decimals
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L157033FND. Rows = 321 Expected = 321.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L157033FPD. Rows = 321 Expected = 321.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: S(7,2). *FORWARDS DESCENDING mthSearch_By_Signed_Decimals
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L57033FNDD. Rows = 321 Expected = 321.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L57033FPDD. Rows = 321 Expected = 321.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: S(7,2). *BACKWARDS DESCENDING mthSearch_By_Signed_Decimals
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L57033FNDD. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L57033FPDD. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  *** SELECT EXACT Key: S(7,2). *FORWARDS mthFunc_Search_Exact_Signed_Decimals
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L157033FND. Rows = 1 Expected = 1.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L157033FPD. Rows = 1 Expected = 1.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: S(7,0). *FORWARDS mthFunc_Search_Generic_Signed
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FND. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FN. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPD. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI1. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI2. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI4. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI8. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: S(7,0). *BACKWARDS mthFunc_Search_Generic_Signed
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FND *BACKWARDS. Rows = 321.  Expected = 321
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FN *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPD *BACKWARDS. Rows = 321.  Expected = 321
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI1 *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI2. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI4. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI8. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: S(7,0). *FORWARDS DESCENDING mthFunc_Search_Generic_Signed
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FNDD. Rows = 642 Expected = 642.
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FNN. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPDD. Rows = 642 Expected = 642.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI1D. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI2D. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI4D. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI8D. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: S(7,0). *BACKWARDS DESCENDING mthFunc_Search_Generic_Signed
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FNDD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FND *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPDD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI1D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI2D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI4D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI8D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  *** SELECT EXACT Key: S(7,0). *FORWARDS mthFunc_Search_Exact_Signed
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FND. Rows = 1 Expected = 1.
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FN. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPD. Rows = 1 Expected = 1.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPA. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI1. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI2. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI4. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI8. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Packed(15,4). *FORWARDS mthSearch_By_Packed_Decimals
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L157033FND. Rows = 0 Expected = 0.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L157033FPD. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Packed(15,4). *BACKWARDS mthSearch_By_Packed_Decimals
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L157033FND. Rows = 0 Expected = 0.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L157033FPD. Rows = 321 Expected = 321.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Packed(15,4). *FORWARDS DESCENDING mthSearch_By_Packed_Decimals
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L57033FNDD. Rows = 0 Expected = 0.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L57033FPDD. Rows = 321 Expected = 321.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Packed(15,4). *BACKWARDS DESCENDING mthSearch_By_Packed_Decimals
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L57033FNDD. Rows = 0 Expected = 0.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L57033FPDD. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  *** SELECT EXACT Key: Packed(15,4). *FORWARDS mthFunc_Search_Exact_Packed_Decimals
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L157033FND. Rows = 1 Expected = 1.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100.00
    2023-12-26 15:16:30 -  On LV L157033FPD. Rows = 1 Expected = 1.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Packed(15,0). *FORWARDS mthFunc_Search_Generic_Decimal
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FND *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FN *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPD *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPA *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI1 *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI2 *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI4 *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI8 *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Packed(15,0). *BACKWARDS mthFunc_Search_Generic_Decimal
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FND *BACKWARDS. Rows = 321.  Expected = 321
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FN *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPD *BACKWARDS. Rows = 321.  Expected = 321
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPA *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI1 *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI2 *BACKWARDS. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI4 *BACKWARDS. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI8 *BACKWARDS. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Packed(15,0). *FORWARDS DESCENDING mthFunc_Search_Generic_Decimal
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FNDD *FORWARDS. Rows = 642 Expected = 642.
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FNN *FORWARDS. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPDD *FORWARDS. Rows = 642 Expected = 642.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPD. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI1D. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI2D. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI4D. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI8D. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Packed(15,0). *BACKWARDS DESCENDING mthFunc_Search_Generic_Decimal
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FNDD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FND *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPDD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI1D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI2D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI4D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI8D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  *** SELECT EXACT Key: Packed(15,0). *FORWARDS mthFunc_Search_Exact_Decimal
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FND. Rows = 1 Expected = 1.
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FN. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPD. Rows = 1 Expected = 1.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPA. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI1. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI2. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI4. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI8. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Int(1). *FORWARDS mthFunc_Search_Generic_Integer1
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FND. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FN. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPD. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPA. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI1. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI2. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI4. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI8. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Int(1). *BACKWARDS mthFunc_Search_Generic_Integer1
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FND *BACKWARDS. Rows = 321.  Expected = 321
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FN *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPD *BACKWARDS. Rows = 321.  Expected = 321
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPA. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI1 *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI2. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI4. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI8. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Int(1). *FORWARDS DESCENDING mthFunc_Search_Generic_Interger1
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FNDD. Rows = 642 Expected = 642.
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FNN. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPDD. Rows = 642 Expected = 642.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPD. Rows = 656 Expected = 656.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI1D. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI2D. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI4D. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI8D. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Int(1). *BACKWARDS DESCENDING mthFunc_Search_Generic_Integer1
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FNDD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FND *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPDD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPD *BACKWARDS. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI1D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI2D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI4D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI8D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  *** SELECT EXACT Key: Int(1). *FORWARDS mthFunc_Search_Exact_Integer1
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FND *FORWARDS. Rows = 1 Expected = 1.
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FN *FORWARDS. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPD *FORWARDS. Rows = 1 Expected = 1.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPA *FORWARDS. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI1 *FORWARDS. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI2 *FORWARDS. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI4 *FORWARDS. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI8 *FORWARDS. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Int(2). *FORWARDS mthFunc_Search_Generic_Integer2
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FND *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FN *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPD *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPA *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI1 *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI2 *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI4 *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI8 *FORWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Int(2). *BACKWARDS mthFunc_Search_Generic_Integer2
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FND *BACKWARDS. Rows = 321.  Expected = 321
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FN *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPD *BACKWARDS. Rows = 321.  Expected = 321
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPA *BACKWARDS. Rows = 320 Expected = 320.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI1 *BACKWARDS. Rows = 336.  Expected = 336
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI2 *BACKWARDS. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI4 *BACKWARDS. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI8 *BACKWARDS. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Int(2). *FORWARDS DESCENDING mthFunc_Search_Generic_Integer2
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FNDD *FORWARDS. Rows = 642 Expected = 642.
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FNN *FORWARDS. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPDD *FORWARDS. Rows = 642 Expected = 642.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPD *FORWARDS. Rows = 656 Expected = 656.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI1D *FORWARDS. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI2D *FORWARDS. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI4D *FORWARDS. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI8D *FORWARDS. Rows = 672 Expected = 672.
    2023-12-26 15:16:30 -  *** SELECT GENERIC Key: Int(2). *BACKWARDS DESCENDING mthFunc_Search_Generic_Integer2
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FNDD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FND *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPDD *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FPD *BACKWARDS. Rows = 336 Expected = 336.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI1D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI2D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI4D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L57033FI8D *BACKWARDS. Rows = 320.  Expected = 320
    2023-12-26 15:16:30 -  *** SELECT EXACT Key: Int(2). *FORWARDS mthFunc_Search_Exact_Integer2
    2023-12-26 15:16:30 -  Key: S(7,2). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FND *FORWARDS. Rows = 1 Expected = 1.
    2023-12-26 15:16:30 -  Key: S(7,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FN *FORWARDS. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,4). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPD *FORWARDS. Rows = 1 Expected = 1.
    2023-12-26 15:16:30 -  Key: DECIMALS(15,0). Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FPA *FORWARDS. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 1. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI1 *FORWARDS. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 2. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI2 *FORWARDS. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 4. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI4 *FORWARDS. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 -  Key: INTEGER 8. Key Value:100
    2023-12-26 15:16:30 -  On LV L157033FI8 *FORWARDS. Rows = 16 Expected = 16.
    2023-12-26 15:16:30 - V57033F. End
    2023-12-26 15:16:30 Completed with <OK>.

ashtkumar2023 commented:
@robe070 , Test Result for :                                                                                                    
1) #Com_Owner.mthSearch_By_Signed_Decimals( #LF_Test_Data #this_rc )
2) #Com_Owner.mthSearch_By_Signed( #LF_Test_Data #this_rc ) : Includes generic and exact Integer2, 4 and 8 results
[Verifier_Test_Report.txt](https://github.com/lansa/db-regression/files/13776024/Verifier_Test_Report.txt)

robe070 commented:
I can't see any updates to the PR. Please do so ASAP.

robe070 commented:
That is, push your git changes to the branch used by the PR.

ashtkumar2023 committed:
IncludesSearch<1> result formthSearch_By_Signed_Decimals and mthSearch_By_Signed

ashtkumar2023 commented:
I pushed again.

robe070 commented on Partition (WBP)/Functions/V57033F.VT157033.fun.yml:
```diff
@@ -183,555 +272,3640 @@
-    #LF_Test_Result_L157033FI1 #LF_Test_Result_L157033FN := 320
-    #LF_Test_Result_L157033FI2 #LF_Test_Result_L157033FI4 #LF_Test_Result_L157033FI8 #LF_Test_Result_L157033FP := 0
-    #LF_Test_Result_L157033FPD #LF_Test_Result_L157033FND := 320
+    #LF_Test_Result_L157033FP := 0
```
You need to implement testing for #LF_Test_Result_L157033FP value too before you complete this particular search.

ashtkumar2023 replied:
Ok..Will be doing it.

ashtkumar2023 replied:
Thank you

robe070 commented on Partition (WBP)/Functions/V57033F.VT157033.fun.yml:
```diff
@@ -183,555 +272,3640 @@
+    * BACKWARDS DESCENDING
+    #LF_Test_Result_L157033FI1 #LF_Test_Result_L157033FI2 #LF_Test_Result_L157033FI4 #LF_Test_Result_L157033FI8 #LF_Test_Result_L157033FND #LF_Test_Result_L157033FPD #LF_Test_Result_L157033FND := 320
+    #LF_Test_Result_L157033FP := 336
+    Add_Entry To_List(#wLTestResultGenericSigned)
```
You are using the same list as on line 277. Thats incorrect. This is still search <1>. Each list only used once per search. I suspect that the entire set of results for BACKWARDS DESCENDING is the same as FORWARDS and hence this whole set of initialisation should be deleted and '#LF_Test_Result_L157033FP := 336' used on line 275.

ashtkumar2023 replied:
Ok..I will look into the above feedback.

ashtkumar2023 replied:
@robe070 , yes. BACKWARDS DESCENDING is the same as FORWARDS in almost all cases . In this case, Can we have 3 lists For Ex:For generic signed results, lists will be:                                                                                                           1) FORWARDS [#wLTestResultGenericSigned]                     
2) BACKWARDS [#wLTestResultGenericSignedBackwards]                                                    
3) FORWARDS DESCENDING [#wLTestResultGenericSignedForwardsDescending]

robe070 replied:
Its not a 'can' question. It depends on the results of a query. As I've posted "Do not duplicate an entire set of results in a new list"

ashtkumar2023 replied:
Search<1> results updated including all numerics. Kindly review .PFA 
[Verifier_Test_Report.txt] (https://github.com/lansa/db-regression/files/13784701/Verifier_Test_Report.txt)
fier report .txt

ashtkumar2023 replied:
Will complete the remaining Search<2> - Search<8>

ashtkumar2023 replied:
Search2 testresults- Working successfully on both IBMi and windows .PFA verifier report. Kindly review.
[Verifier_Test_Report.txt](https://github.com/lansa/db-regression/files/13790972/Verifier_Test_Report.txt)

robe070 replied:
So far so good.

robe070 reviewed (changes requested):
Fix up these items.
Implement Search<2> test results.
Don't require me to find these same types of issues again.
That is, 

1. Each column in the table needs to have a value set in the results.
2. Only use a list once per Search<n>
3. Do not duplicate an entire set of results in a new list as often happens between *FORWARDs and *BACKWARDS DESCENDING

ashtkumar2023 committed:
Search<1> results for all Logical Views

ashtkumar2023 committed:
Search 2 testdata and results-fixed on both IBMi and WIndows standalone

ashtkumar2023 committed:
Issue#30 -Complete PR implementation

ashtkumar2023 commented:
[Verifier_Test_Report.txt](https://github.com/lansa/db-regression/files/13810973/Verifier_Test_Report.txt)
The report for the all the numerics. PFA

robe070 commented on Partition (WBP)/Functions/V57033F.VT157033.fun.yml:
```diff
@@ -528,210 +931,4942 @@
+    #LF_View_Type := ' Key: INTEGER 2'
+    #Com_Owner.mthAdd_Msg( #this_rc (#LF_View_Type) ('Key Value:' + #this_test_value.asString) )
+    
+    #LF_Test_Data_INT2 := #this_test_value
```
#this_test_value MUST be used as the key in the SELECT unless there is an error. In practice that means that a Field with greater length cannot be used with a key of lesser length. E.g. A field of INT 2 cannot be used directly as a key of INT 1. Only in this circumstance is your existing code correct. All uses of this construct must be corrected.

ashtkumar2023 replied:
I have incorporated the above review comment and ran in Windows standalone successfully and is running in IBMi still. Once it is done, will share the verifier test report. Could you please review the code ?

robe070 reviewed (changes requested):
All looks good apart from the comment I've made.

ashtkumar2023 committed:
#this_test_value MUST be used as the key in the SELECT

robe070 reviewed (changes requested):
Most of the time you are leaving this code in, unchanged or commented out. Delete it.
```
#LF_Test_Data_INT2 := #this_test_value
```

ashtkumar2023 committed:
Removed the comment or the unwanted code

ashtkumar2023 commented:
I have removed them from the places where I had retained it as mentioned above.
Also,I have replaced LF_Test_Data_INT2 with #this_test_value wherever there were no errors. .
Hence this line is retained in few places.PFA
#LF_Test_Data_INT2 := #this_test_value 
![image](https://github.com/lansa/db-regression/assets/147069578/39980e09-c659-4093-a2c2-e85c30018d97)
Verifier report : 
[Verifier_Test_Report.txt](https://github.com/lansa/db-regression/files/13819729/Verifier_Test_Report.txt)

robe070 reviewed (changes requested):
You've changed 1 line of the kind I was talking about? There are probably more than 20!
Search for "#LF_Test_Data_INT"

Most of the hits of that search where the Field is assigned from #this_test_value need to be changed, along with the SELECT.

ashtkumar2023 commented:
I will use the Integer Intrinsic function 'AsValue' to pass the #this_test_value as a Key in the  SELECT 
![image](https://github.com/lansa/db-regression/assets/147069578/3b25a592-7e45-4817-8606-da6dd990e937)
I will modify the same for Integer 1, Integer4, Integer8 as well.

ashtkumar2023 committed:
passing #this_test_value in SELECT

ashtkumar2023 commented:
I have incorporated the above comment. PFA the report on windows. IBMi test is still running.
[Verifier_Test_Report.txt](https://github.com/lansa/db-regression/files/13827484/Verifier_Test_Report.txt)

ashtkumar2023 commented:
[Verifier_Test_Report.txt](https://github.com/lansa/db-regression/files/13829094/Verifier_Test_Report.txt)
Complete Verifier report
